python/xorbits/_mars/dataframe/merge/merge.py

```diff
@@ -855,26 +872,70 @@ def execute(cls, ctx, op):
                 right = right.iloc[index]
 
         def execute_merge(x, y):
+            """
+            All merge and join operations are transformed into specific merge operations.
+            In particular, for join operations on the GPU without the ``on`` column
+            (CUDF does not support join with ``on`` parameter),
+            The key of merge is always transformed into a column on one side (left_on / right_on) and
+            an index on the other side (left_index=True / right_index=True).
+            In this case, CUDF takes the result of the ``on`` column as the index instead of the data column,
+            which is different from pandas.
+            so we should reset the index before the real calculation, and
+            then restore the column to the ``on`` data column.
+            """
+            left_index = None
```

Comment:
These variables are only used for GPU, so I guess they should be put into the `else` branch.

Reply:
These variables are used in line 910, and in the case of CPU they are always None to distinguish the corresponding logic.